src/tokenomics/get-history-events.ts

```diff
@@ -0,0 +1,156 @@
+  const slash: Slashing = {
+    validatorStash: validatorStash.toString(),
+    slashedAmount: slashedAmount.toNumber(),
+    blockHeight: blockHeight,
+    session: session.toNumber(),
+    era: era.toNumber()
+  };
+  return slash
+}
+
+export async function getProposalCreated(api: ApiPromise, proposalId: ProposalId, blockHash:Hash): Promise<ActiveProposal> {
+  const proposalData = await api.query.proposalsEngine.proposals.at(blockHash,proposalId) as Proposal;
+  const proposalStatus = proposalData.get("status") as ProposalStatus
+  const proposalsDetails = await api.query.proposalsCodex.proposalDetailsByProposalId.at(blockHash,proposalId) as ProposalDetails
+  const activeProposal: ActiveProposal = {
+    id: proposalId.toNumber(),
+    type: proposalsDetails.type.toString(),
+    title: proposalData.get("title")?.toString(),
+    createdBy: (proposalData.get("proposerId") as MemberId).toNumber(),
+    created: (proposalData.get("createdAt") as BlockNumber).toNumber(),
+    statusAtStart: proposalStatus.type,
+    voters: [],
+    status: proposalStatus.type,
+  };
+  if (proposalsDetails.value instanceof SpendingParams) {
+    activeProposal.spending = (proposalsDetails.value[0] as Balance).toNumber()
+  }
+  if (proposalStatus.value instanceof Active) {
+    const stakeId = ((proposalStatus.value.value) as ActiveStake).get("stakeId") as StakeId
+    const stakeData = (await api.query.stake.stakes.at(blockHash,stakeId) as Tuple)[0] as Stake
+    activeProposal.stakeId = stakeId.toNumber()
+    activeProposal.stake = (stakeData.staking_status.value as Staked).staked_amount.toNumber()
+  }
+  return activeProposal
+}
+
+export async function getVote(event: EventData, blockHeight: number): Promise<Voters> {
+  const memberId = event[0] as MemberId;
+  const voteKind = event[2] as VoteKind;
+  const vote: Voters = {
+    voter: memberId.toNumber(),
+    vote: voteKind.toString(),
+    blockNumber: blockHeight
+  };
+  return vote
+}
+
+export async function getProposalStatusUpdated(event: EventData, proposals: ActiveProposal[], proposalIds: number[], blockHeight: number): Promise<ActiveProposal[]> {
+  const proposalId = event[0] as ProposalId
+  const proposalStatus = event[1] as ProposalStatus
+  if (proposalIds.includes(proposalId.toNumber())) {
+    const index = proposalIds.indexOf(proposalId.toNumber())
+    if (proposalStatus.value instanceof Finalized) {
+      proposals[index].status = proposalStatus.type
+      proposals[index].finalizedAt = blockHeight
+      const decision = proposalStatus.value.get("proposalStatus") as ProposalDecisionStatus
+      proposals[index].result = decision.type
+      if (decision.value instanceof Approved) {
+        proposals[index].executionStatus = decision.value.type
+        if (decision.value.toString()!=="PendingExecution") {
+          proposals[index].executedAt = blockHeight
+        }
+      } else {
+        if (decision.type === "Canceled" ||decision.type === "Cancelled") {
```

Comment:
I thought `Canceled` was incorrect (one "l"), so I figured it may be changed in the future :)

A quick search uncovered that it's one of the US/British English differences, so it makes less sense!